packages/image_picker/image_picker_ios/ios/Classes/FLTImagePickerPhotoAssetUtil.m

```diff
@@ -20,6 +20,17 @@ + (PHAsset *)getAssetFromPHPickerResult:(PHPickerResult *)result API_AVAILABLE(i
   return fetchResult.firstObject;
 }
 
++ (NSURL *)saveVideoFromURL:(NSURL *)videoURL error:(NSError **)error {
+  if (![[NSFileManager defaultManager] isReadableFileAtPath:[videoURL path]]) {
+      *error = [NSError errorWithDomain:@"FLTImagePicker" code:9999 userInfo:@{ NSLocalizedDescriptionKey : @"No readable file af provided path" }];
+    return videoURL;
```

Comment:
I don't know if this error is necessary, if so, I could use some feedback on best practices.

Reply:
Given that the caller is throwing away the details of the NSError anyway, I would just have the function return `nil` when either call fails.

Alternately, you could make the calling method use the NSError to build a FlutterError. If you go that route, this seems reasonable, but I would still return `nil` on error. I would also make the code less random (`e.g.`, 1), and you'd want to use [`NSLocalizedDescriptionKey`](https://developer.apple.com/documentation/foundation/nslocalizeddescriptionkey?language=objc) for the details key (and fix the typo in the message).